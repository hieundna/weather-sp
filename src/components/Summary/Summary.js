import '../../styles/Summary.scss';
import Header from '../Header/Header';
import { useAppContext } from '../../contexts/context'
import { getWeatherData } from '../../api/index'
import { getWeatherInfor } from '../../contexts/actions/action'
import { useEffect } from 'react';
import WeatherStatus from '../WeatherStatus/WeatherStatus';
import WeatherDetail from '../WeatherDetail/WeatherDetail';

function Summary() {

    const dispatch = useAppContext()[1]
    useEffect(() => {
        const getData = async () => {
            try {
                let data = await getWeatherData('Singapore');
                dispatch(getWeatherInfor(data.data));
            } catch (error) {
                console.error('Error get data: ', error);
            }
        }
        getData()
    }, [])

    return (
        <div role='summary' className='weather-sumary'>
            <Header />
            <WeatherStatus />
            <WeatherDetail />
        </div>
    );
}

export default Summary;
