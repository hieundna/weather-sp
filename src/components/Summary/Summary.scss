.weather-sumary {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 65vh;
}