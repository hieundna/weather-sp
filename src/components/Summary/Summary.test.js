import React from 'react';
import Summary from './Summary.js';
import { render } from '@testing-library/react';

const setup = () => render(<Summary />);
describe('Summary Component test', () => {
    test("expect to render Summary component", () => {
        const wrapper = setup();
        const title = wrapper.getByRole('summary');
        expect(title).toHaveClass('weather-sumary');
        expect(wrapper).toMatchSnapshot();
    })
})