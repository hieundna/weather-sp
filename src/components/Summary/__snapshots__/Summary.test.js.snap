// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Component test expect to render Summary component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="weather-sumary"
        role="summary"
      >
        <div
          class="header"
        >
          <div
            class="burger"
          >
            <span />
            <span />
            <span />
          </div>
          <div
            class="title"
          >
            <div
              role="myENV-test"
            >
              myENV
            </div>
            <div
              class="list-city"
            >
              <span
                class="city-name"
              >
                Singapore
              </span>
              <i
                class="fal fa-angle-down fa-lg"
              />
            </div>
            <div
              class="dropdown "
            >
              <div
                class="option"
              >
                Ho Chi Minh
              </div>
              <div
                class="option"
              >
                Vinh
              </div>
              <div
                class="option"
              >
                Singapore
              </div>
            </div>
          </div>
          <div>
            <i
              class="fas fa-bell fa-2x"
            />
          </div>
        </div>
        <div
          class="main"
        >
          <div
            class="state-icon"
          >
            <img
              alt="cloud-icon"
              src="http://openweathermap.org/img/wn/03n@2x.png"
            />
          </div>
          <div
            class="main-title"
          >
            <div
              class="state"
            >
              Clouds
            </div>
            <div
              class="detail"
            >
              <i
                class="fas fa-thermometer-half"
              />
              <span>
                28
                °C
              </span>
            </div>
          </div>
          <div
            class="detail"
          >
            <i
              class="far fa-tint"
            />
            <span>
              83
              %
            </span>
          </div>
        </div>
        <div
          class="weather-detail"
        >
          <div
            class="col-detail"
          >
            <div
              class="title-detail"
            >
               PSI
            </div>
            <div
              class="detail-value psi"
            >
              32
            </div>
            <div
              class="detail-state"
            >
              Good
            </div>
          </div>
          <div
            class="col-detail"
          >
            <div
              class="title-detail"
            >
               RAIN
            </div>
            <div
              class="detail-value"
            >
              0
            </div>
            <div
              class="detail-state"
            >
              mm
            </div>
          </div>
          <div
            class="col-detail"
          >
            <div
              class="title-detail"
            >
               DENGUE
            </div>
            <div
              class="circle-icon"
            />
          </div>
          <div
            class="col-detail"
          >
            <div
              class="add-icon"
            >
              <i
                class="far fa-plus-circle fa-2x"
              />
            </div>
            <div
              class="add-text"
            >
              Add
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weather-sumary"
      role="summary"
    >
      <div
        class="header"
      >
        <div
          class="burger"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          class="title"
        >
          <div
            role="myENV-test"
          >
            myENV
          </div>
          <div
            class="list-city"
          >
            <span
              class="city-name"
            >
              Singapore
            </span>
            <i
              class="fal fa-angle-down fa-lg"
            />
          </div>
          <div
            class="dropdown "
          >
            <div
              class="option"
            >
              Ho Chi Minh
            </div>
            <div
              class="option"
            >
              Vinh
            </div>
            <div
              class="option"
            >
              Singapore
            </div>
          </div>
        </div>
        <div>
          <i
            class="fas fa-bell fa-2x"
          />
        </div>
      </div>
      <div
        class="main"
      >
        <div
          class="state-icon"
        >
          <img
            alt="cloud-icon"
            src="http://openweathermap.org/img/wn/03n@2x.png"
          />
        </div>
        <div
          class="main-title"
        >
          <div
            class="state"
          >
            Clouds
          </div>
          <div
            class="detail"
          >
            <i
              class="fas fa-thermometer-half"
            />
            <span>
              28
              °C
            </span>
          </div>
        </div>
        <div
          class="detail"
        >
          <i
            class="far fa-tint"
          />
          <span>
            83
            %
          </span>
        </div>
      </div>
      <div
        class="weather-detail"
      >
        <div
          class="col-detail"
        >
          <div
            class="title-detail"
          >
             PSI
          </div>
          <div
            class="detail-value psi"
          >
            32
          </div>
          <div
            class="detail-state"
          >
            Good
          </div>
        </div>
        <div
          class="col-detail"
        >
          <div
            class="title-detail"
          >
             RAIN
          </div>
          <div
            class="detail-value"
          >
            0
          </div>
          <div
            class="detail-state"
          >
            mm
          </div>
        </div>
        <div
          class="col-detail"
        >
          <div
            class="title-detail"
          >
             DENGUE
          </div>
          <div
            class="circle-icon"
          />
        </div>
        <div
          class="col-detail"
        >
          <div
            class="add-icon"
          >
            <i
              class="far fa-plus-circle fa-2x"
            />
          </div>
          <div
            class="add-text"
          >
            Add
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
